### DISCUSSION TITLE
How to hook events when a client stream disconnects or network error?

### DISCUSSION THREAD

**Discussion by RustChater** [hyperium/tonic#864] [Q&A]
I want to do something when client stream disconnects or network error.
the definition of proto like this:
```
rpc MsgPush (MsgPushReq) returns (stream MsgsRes) {}
```

rust related code:
```
type MsgPushStream = tokio_stream::wrappers::ReceiverStream<Result<MsgsRes, Status>>;
```

**Comment by LucioFranco** [comment]
So I think when you try to write a message on the write end and it fails with a closed that means the receiver went away and the client has disconnected. So you can use that to detect that.
